### DISCUSSION TITLE
Caching headers are presented in Dev mode while missing in Prod mode

### DISCUSSION THREAD

**Discussion by SZharkov** [vercel/next.js#64630] [Ideas]
# Bug report

## Describe the bug
`Cache-control` header is presented in response when working in `dev` mode. In `prod` mode this header is missing. I was also trying to set headers manually using custom express server before sending a response `res.set("Cache-control", "no-store, must-revalidate")` but it didn't make any change.

## To Reproduce
1. `yarn create next-app`
2. `yarn dev` - check response headers
3. `yarn build`, `yarn start` - check response headers

## Expected behavior
`Cache-control` header is presented in both `dev` and `prod` environments.

## Screenshots
![image](https://user-images.githubusercontent.com/13063421/76617831-c7500580-652f-11ea-9ff6-76a19bcab503.png)
![image](https://user-images.githubusercontent.com/13063421/76617984-1eee7100-6530-11ea-8e1e-eeada3cf1671.png)

**Comment by valmassoi** [comment]
I'm having the same issue. `X-Cache-Expired-At` and `X-Cache-Status`(from ssr-caching example - cacheable-response package) are there locally in dev mode but missing when deployed to prod. Did you figure something out @SZharkov 

I dropped back to version `9.3.0` to fix this
https://github.com/vercel/next.js/issues/12019

**Comment by andres-dejesus** [comment]
Just opened https://github.com/vercel/next.js/issues/40275 which looks similar, its been 2 years, any progress?
